src/ai/backend/common/events.py

```diff
@@ -1435,14 +1437,17 @@ async def close(self) -> None:
     async def produce_event(
         self,
         event: AbstractEvent,
-        *,
-        source: str = "manager",
+        source_override: Optional[AgentId] = None,
```

Comment:
t seems that the source should be able to include components other than just Agent.

Reply:
How about defining and using a `NodeId` that includes an `AgentId`?

Reply:
Since the existing behavior is taking the agentId, I wanted to avoid modifying the event producer side as much as possible.